gio/src/subclass/list_store.rs

```diff
@@ -0,0 +1,13 @@
+use glib::subclass::prelude::*;
+
+use crate::ListStore;
+use glib::subclass::object::ObjectImpl;
+use glib::Object;
+
+pub trait ListStoreImpl: ObjectImpl {}
+
+unsafe impl<T: ListStoreImpl> IsSubclassable<T> for ListStore {
+    fn override_vfuncs(class: &mut glib::Class<Self>) {
+        <Object as IsSubclassable<T>>::override_vfuncs(class);
```

Comment:
This is not actually sub-classable :)

```
G_DECLARE_FINAL_TYPE(GListStore, g_list_store, G, LIST_STORE, GObject)
```

There's no public class struct. How does this currently compile at all? The `GListStoreClass` struct seems to only exist in the .gir file but not in the headers. That seems concerning and should probably be fixed (by removing it from the .gir file!). Can you create an issue for that (against glib) and remove it from here?

Reply:
Ah my bad. I knew you cannot subclass it (from Vala days) but the generated class struct confused me a I guess. Opened an issue https://gitlab.gnome.org/GNOME/glib/-/issues/2261

Reply:
That probably also means that we need to extend the "final type" heuristic in gir to consider instance structs without any fields at all as indicator. Can you create an issue for that?

Reply:
https://github.com/gtk-rs/gir/issues/1013

Reply:
That actually means there's some widgets/objects that shouldn't be really subclassable... I wonder how Vala or other languages do check that?

Reply:
You mean widgets that we already allow to subclass but that shouldn't? For Vala it would fail to compile because there is no instance struct in the C headers (because Vala goes via C in the end), same for the C++ (mm) bindings. Others probably just don't care? :) Try subclassing `GtkListBox` from Python maybe, but I'm 99% sure it actually works.

Reply:
Which other widgets btw?

Also I think if you would try actually subclassing `ListBox` from Rust the way you had it before, you would likely get a critical warning at runtime because we would create a too small C instance struct (as the actual one is not known from the .gir file).

Reply:
> Which other widgets btw?
> 
> Also I think if you would try actually subclassing `ListBox` from Rust the way you had it before, you would likely get a critical warning at runtime because we would create a too small C instance struct (as the actual one is not known from the .gir file).

I don't think there are any currently, at least not in gtk-rs/gtk4-rs, I will have to check libhandy stuff though.